packages/functional-tests/pages/signinReact.ts

```diff
@@ -0,0 +1,75 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { expect } from '@playwright/test';
+import { BaseLayout } from './layout';
+import { getReactFeatureFlagUrl } from '../lib/react-flag';
+
+export class SigninReactPage extends BaseLayout {
+  readonly path = 'signin';
+
+  get authenticationFormHeading() {
+    return this.page.getByRole('heading', {
+      name: /^Enter (?:authentication|security) code/,
+    });
+  }
+
+  get authenticationCodeTextbox() {
+    return this.page
+      .getByRole('textbox', { name: 'code' })
+      .or(this.page.getByPlaceholder('Enter 6-digit code'));
+  }
+
+  get authenticationCodeTextboxTooltip() {
+    return this.page.getByText('Invalid two-step authentication code', {
+      exact: true,
+    });
+  }
+
+  get confirmButton() {
+    return this.page.getByRole('button', { name: 'Confirm' });
+  }
+
+  get passwordFormHeading() {
+    return this.page.getByRole('heading', { name: /^Enter your password/ });
+  }
+
+  get passwordTextbox() {
+    return this.page.getByRole('textbox', { name: 'password' });
+  }
+
+  get signInButton() {
+    return this.page.getByRole('button', { name: 'Sign in' });
+  }
+
+  goto(route = '/', params = new URLSearchParams()) {
+    params.set('forceExperiment', 'generalizedReactApp');
+    params.set('forceExperimentGroup', 'react');
+    return this.page.goto(
+      getReactFeatureFlagUrl(this.target, route, params.toString())
+    );
+  }
+
+  async fillOutAuthenticationForm(code: string): Promise<void> {
+    await expect(
+      this.authenticationFormHeading,
+      'The Authentication form heading is missing or not as expected. ' +
```

Comment:
👀 Very nice to get a clear error message when this assertion fails (noting the pattern for reuse.....). [Based on the docs](https://playwright.dev/docs/test-assertions#custom-expect-message) I would expect to see the custom message to be shown for both passing and failing assertions, but I only see it shown on error. Is that due to our test configuration?

Reply:
Oh! Thanks for pointing this out! I didn't realize the message on an expect statement would be output by default in both cases, that's not the behavior I'm used to with assertions. 

I also don't know why they aren't appearing in the report, like you mentioned:
<img width="1227" alt="image" src="https://github.com/mozilla/fxa/assets/3422688/4a3c63c3-b7f4-491d-9c70-7f34a5a2e265">


To be safe I'll remove them though, because if they start outputting that could be very cluttered and I don't want to set a bad precedent  😬

Reply:
I see that for Playwright for Python, it's described as a custom error message in the same context... and the docs for Node were recently updated to the current state where it's a pass/fail message (yet we're not seeing it in the report....). Confusing. Perhaps this changed between our current version of playwright and latest.

Error message as you've used it would have been more useful. Too bad!

Reply:
I think you may be right about this being due to version.